utils/tolerance.stanza

```diff
@@ -3,36 +3,56 @@ defpackage ocdb/tolerance:
+; tol+/tol- are relative to typ and not absolute
 public pcb-struct ocdb/tolerance/Toleranced:
   typ: Double,
   tol+:Double,
   tol-:Double,
 
 defmethod print (o:OutputStream, t:Toleranced):
-  print(o, "Toleranced(%_ <= %_ <= %_)" % [min-value(t), typ(t), max-value(t)])
+  print(o, "Toleranced(%_ <= %_ <= %_)" % [min-value(t), typ-value(t), max-value(t)])
 
 ; Create a tolerance with minimum, typical, and maximum
-public defn min-typ-max (min:Double, typ:Double, max:Double) -> Toleranced:
-  val tol- = typ - min
-  val tol+ = max - typ
-  tol(typ, tol+, tol-)
+public defn min-typ-max (min:Double, typ:Double|False, max:Double) -> Toleranced:
+  val typ-r = match(typ) :
+    (t:Double) : t
+    (f) : min + 0.5 * (max - min)
+  val tol- = typ-r - min
+  val tol+ = max - typ-r
+  tol(typ-r, tol+, tol-)
 
-; Create a tolerance from percentages higher and lower
+; Create a tolerance with minimum, and maximum 
+; with the typ assumed to be in the center of the range of {min, max}
+public defn min-max (min:Double, max:Double) -> Toleranced:
+  min-typ-max(min, false, max)
+
+; Create a tolerance from percentages higher and lower 
+; where the argument is the percentage (double) in the range of 0.0 to 100.0
+; and not a decimal representation of the percentage.
 public defn tol% (typ:Double, tol+:Double, tol-:Double) -> Toleranced:
-  val check = min-typ-max(0.0, 0.5, 1.0)
-  fatal("tol+ must be in range 0.0 <= tol+ <= 1.0") when not in-range?(check, tol+)
-  fatal("tol- must be in range 0.0 <= tol- <= 1.0") when not in-range?(check, tol-)
-  val plus  = tol+ * typ
-  val minus = tol- * typ
+  val check100 = min-typ-max(0.0, false, 100.0)
+  val check1 = min-typ-max(0.0, false, 1.0)
+  fatal("tol+ must be in range 0.0 <= tol+ <= 100.0") when not in-range?(check100, tol+)
+  fatal("tol- must be in range 0.0 <= tol- <= 100.0") when not in-range?(check100, tol-)
+  warning("tol%(typ, tol+, tol-) -> tol+ might be mis-specified (< 1.0), expecting 0.0 <= tol+ <= 100.0") when in-range?(check1, tol+)
+  warning("tol%(typ, tol+, tol-) -> tol- might be mis-specified (< 1.0), expecting 0.0 <= tol- <= 100.0") when in-range?(check1, tol-)
+  val plus  = typ * tol+ / 100.0
+  val minus = typ * tol- / 100.0
   Toleranced(typ, plus, minus)
 
-; Create a tolerance from percentage
+; Create a symmetric tolerance from percentage 
+; where the percentage ranges from 0.0 to 100.0 (not 0.0 to 1.0)
 public defn tol% (typ:Double, tol:Double) -> Toleranced:
-  val check = min-typ-max(0.0, 0.5, 1.0)
-  fatal("tol must be in range 0.0 <= tol <= 1.0") when not in-range?(check, tol)
-  Toleranced(typ, typ * tol, typ * (-1.0 * tol))
+  val check100 = min-typ-max(0.0, false, 100.0)
+  val check1 = min-typ-max(0.0, false, 1.0)
+  fatal("tol must be in range 0.0 <= tol <= 100.0") when not in-range?(check100, tol)
+  warning("tol%(typ, tol) -> tol might be mis-specified (< 1.0), expecting 0.0 <= tol <= 100.0") when in-range?(check1, tol)
```

Comment:
see the comment above, I don't think we need a warning here.

Reply:
We absolutely need a warning to the user -- if they use 0.1 as the argument, did they mean 10% or 0.1%? More often it will be 10% but we need to hold our user's hands.

Reply:
So why not leave the original implementation?